packages/agents-core/src/tool.ts

```diff
@@ -110,6 +111,61 @@ export function computerTool(
   };
 }
 
+export type HostedMCPApprovalFunction<Context = UnknownContext> = (
+  context: RunContext<Context>,
+  data: RunToolApprovalItem,
+) => Promise<{ approve: boolean; reason?: string }>;
+
+/**
+ * A hosted MCP tool that lets the model call a remote MCP server directly
+ * without a round trip back to your code.
+ */
+export type HostedMCPTool<Context = UnknownContext> = HostedTool & {
+  name: 'hosted_mcp';
+  providerData: ProviderData.HostedMCPTool<Context>;
+};
+
+/**
+ * Creates a hosted MCP tool definition.
+ *
+ * @param serverLabel - The label identifying the MCP server.
+ * @param serverUrl - The URL of the MCP server.
+ * @param requireApproval - Whether tool calls require approval.
+ */
+export function hostedMcpTool<Context = UnknownContext>(
+  options: {
+    serverLabel: string;
+    serverUrl: string;
+  } & (
+    | { requireApproval?: 'never' }
+    | {
+        requireApproval: 'always';
+        onApproval: HostedMCPApprovalFunction<Context>;
+      }
+  ),
+): HostedMCPTool<Context> {
+  const providerData: ProviderData.HostedMCPTool<Context> =
+    options.requireApproval === 'always'
+      ? {
+          type: 'mcp',
+          serverLabel: options.serverLabel,
+          serverUrl: options.serverUrl,
+          requireApproval: 'always',
+          onApproval: options.onApproval,
+        }
+      : {
+          type: 'mcp',
+          serverLabel: options.serverLabel,
+          serverUrl: options.serverUrl,
+          requireApproval: 'never',
```

Comment:
lack of this property results in 400 bad request error, so explicitly having this